packages/go_router/test/delegate_test.dart

```diff
@@ -56,6 +57,34 @@ void main() {
     });
   });
 
+  group('push', () {
+    testWidgets(
+      'It should return different pageKey when push is called',
+      (WidgetTester tester) async {
+        final GoRouter goRouter = await createGoRouter(tester);
+        expect(goRouter.routerDelegate.matches.matches.length, 1);
+        expect(
+          goRouter.routerDelegate.matches.matches[0].pageKey,
+          null,
+        );
+
+        goRouter.push('/a');
```

Comment:
can you do an `await tester.pumpAndSettle()` in between every push to verify it does not crash?

Reply:
Thanks for the review, I fixed it in 2ee5fd6.